### PR TITLE
feat: scan isbn and confirm to autofill to book

### PR DESCRIPTION
**Pull request by c25vdw**
integrated with add/edit book activity

### CODE REVIEW

**Comment by c25vdw** [comment]
<details><summary>demo (youtube link)</summary>

[](https://youtu.be/alOz-6afiOI)
</details>

File: app/src/main/java/com/cmput301f20t21/bookfriends/ui/scanner/ScannerBaseActivity.java [hunk 1]
```diff
@@ -27,14 +28,15 @@
  * the base Activity to be extended by any that needs the scanning functionality
  */
 public class ScannerBaseActivity extends AppCompatActivity {
+    public static final String ISBN_KEY = "ISBN_KEY";
```
Comment (zhpeng811) [inline app/src/main/java/com/cmput301f20t21/bookfriends/ui/scanner/ScannerBaseActivity.java:31]: I think we are using the convention that all keys should have the `com.cmput301f20t21.bookfriends` prefix

other than that it lgtm

**Review by zhpeng811 — APPROVED** [review]

File: app/src/main/java/com/cmput301f20t21/bookfriends/ui/add/AddEditActivity.java [hunk 2]
```diff
@@ -34,6 +35,7 @@
     private ImageView bookImage;
     private Uri bookImageUri;
     private TextInputLayout isbnLayout;
+    private EditText isbnTextEdit;
```
Comment (kdvuong) [inline app/src/main/java/com/cmput301f20t21/bookfriends/ui/add/AddEditActivity.java:38]: wrong name, should be isbnEditText.

**Review by kdvuong — APPROVED** [review]
Would it be better user experience if we have a dialog pop up for user to confirm when the bar code is scanned (instead of reaching for the confirm in the corner)? PR looks good tho, this could be added in the future.

**Pull request merged**